implementations/micrometer-registry-elastic/src/main/java/io/micrometer/elastic/ElasticMeterRegistry.java

```diff
@@ -165,22 +169,27 @@ private String indexName() {
 
     // VisibleForTesting
     Optional<String> writeCounter(Counter counter) {
-        return Optional.of(writeDocument(counter, builder -> {
-            builder.append(",\"count\":").append(counter.count());
-        }));
+        return writeCounter(counter, counter.count());
     }
 
     // VisibleForTesting
     Optional<String> writeFunctionCounter(FunctionCounter counter) {
-        return Optional.of(writeDocument(counter, builder -> {
-            builder.append(",\"count\":").append(counter.count());
-        }));
+        return writeCounter(counter, counter.count());
+    }
+
+    private Optional<String> writeCounter(Meter meter, Double value) {
+        if (Double.isFinite(value)) {
```

Comment:
@fkoehler Could counters also be NaN or infinity? If so, other meter registries which handle NaN and infinity gauges should also handle NaN and infinity counters (not in this PR).

Reply:
`FunctionCounters` could be. Regular `Counters` could not I think because of the signature `void increment(double amount)`.

Reply:
As @jkschneider said, I found this while looking at the kafka metrics and sending them to elastic. It happens for function counters. So other registries should check for that as well.

Reply:
I created #1138 for other meter registries.